src/main/java/com/google/devtools/build/lib/analysis/test/TestRunnerAction.java

```diff
@@ -531,6 +531,11 @@ public void setupEnvVariables(Map<String, String> env, Duration timeout) {
     }
     env.put("XML_OUTPUT_FILE", getXmlOutputPath().getPathString());
 
+    if (!isEnableRunfiles()) {
```

Comment:
don't we need this change for non-test actions as well?

Reply:
I think this is also fixed in native launcher, https://github.com/bazelbuild/bazel/blob/1b3c6816eb82a07e0e5973e0e7ace1e309fd71fc/src/tools/launcher/launcher.cc#L231-L241